src/bigdecimal.ts

```diff
@@ -4350,6 +4350,37 @@ export class BigDecimal {
             return BigDecimal.divideAndRound3(dividend, scaledDivisor, scale, roundingMode, scale);
         }
     }
+
+    /**
+     * Function called in case BigDecimal is implicitly converted to a primitive value.
+     * The function will throw when the code try to convert the BigDecimal value in the following cases:
+     * - Using a greater or lower comparison operators (>, <, <=, >=)
+     * - Using loose equality operators against number, string, symbol or bigint (==, !=)
+     * - Using an arithmetic operators (+, -, /, *, %, **, --, ++)
+     * - Using bitwise operators (&, |, ^, ~, >>, >>>, <<)
+     *
+     * Note:
+     * - In case of string templates, the function will return the value from {@link toString}.
+     * - It will throw when the plus (+) sign is used for addition or concatenation.
```

Comment:
I think comparison operators will also make our function throw because 'number' hint is passed as I understand from the documentation. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Less_than 

Is there any other operation that can throw as well? Should we enumerate all of them in this Note? WDYT?

Also, concatenation will work like this, maybe we can remove the "it will throw when" statement? 

""+ Big("1.2123123")


I mean, because listing all the situations where we can throw seems hard to me. Even with just addition operator, only some cases throw exception. (e.g Big("1.2") + Big("2.1") )